### ISSUE TITLE
Application built with pyinstaller does not start

### ISSUE THREAD

**Issue by rejjin** [dashingsoft/pyarmor#32]
Hey. How can I fix this issue?

I have simple script:

![image](https://user-images.githubusercontent.com/692007/46903448-aff58e80-cedd-11e8-9a38-e81587e8f187.png)

I obfuscate the script:
`pyarmor obfuscate --src . -O obfuscated --entry start.py`

![image](https://user-images.githubusercontent.com/692007/46903463-f1863980-cedd-11e8-8cbb-398f10912331.png)

Now I build with PyInstaller:

`pyinstaller --noconfirm --log-level=WARN --onefile --distpath=. --clean --paths . "./obfuscated/start.py"
`

And I get an error while running the built application:

> > start.exe
> Traceback (most recent call last):
>   File "site-packages\PyInstaller\loader\pyiboot01_bootstrap.py", line 149, in __init__
>   File "ctypes\__init__.py", line 348, in __init__
> OSError: [WinError 126] Не найден указанный модуль
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "pytransform.py", line 207, in _load_library
>   File "ctypes\__init__.py", line 426, in LoadLibrary
>   File "site-packages\PyInstaller\loader\pyiboot01_bootstrap.py", line 151, in __init__
> __main__.PyInstallerImportError: Failed to load dynlib/dll 'C:\\Temp\\_MEI87362\\_pytransform.dll'. Most probably this dynlib/dll was not found when the application was frozen.
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "start.py", line 2, in <module>
>     print('result: {sum}'.format(sum=x*y))
>   File "pytransform.py", line 239, in pyarmor_runtime
>   File "pytransform.py", line 233, in pyarmor_init
>   File "pytransform.py", line 213, in _load_library
> pytransform.PytransformError: ('Could not load _pytransform from "%s"', 'C:\\Temp\\_MEI87362')
> [4644] Failed to execute script start

Here is the contents of the application directory:
_pytransform.dll  build  license.lic  obfuscated  product.key  pyshield.key  pyshield.lic  start.exe  start.py

**Comment by rejjin** [comment]
Fixed with declarate lib path

![image](https://user-images.githubusercontent.com/692007/46903717-6e66e280-cee1-11e8-9ea6-0984c0779ad0.png)

**pyinstaller --noconfirm --log-level=WARN --onefile --distpath=. --clean --paths "./lib" "./obfuscated/start.py"**

But why doesn't the application search the current directory for the necessary files?

**Comment by jondy** [comment]
By default, pyarmor search the necessary files in the path of file `pytransform.py`

https://github.com/dashingsoft/pyarmor/blob/master/src/pytransform.py#L189

```
def _load_library(path=None):
    if path is None:
        path = os.path.dirname(__file__)
```

Most of cases, it's fine. But for pyinstaller and py2exe, it will package python scripts to a zip file. So pyarmor can not find the necessary files by attribute `__filename__` of module `pytranform`.

**Comment by rejjin** [comment]
Thank you very much, it's working fine now.

**Comment by deeuu** [comment]
@rejjin or @jondy Can you elaborate on this?

I've obfuscated my package, but not sure whether I need to include the `_pytransform.so` in the spec file used by pyinstaller. Currently, pyinstaller can't find the shared library....

**Comment by jondy** [comment]
```
    # First create project
    #
    cd /path/to/pyarmor
    python pyarmor.py init --src=examples/py2exe \
                           --entry="hello.py" \
                           projects/py2exe
    cd projects/py2exe

    # This is the key, change default `runtime-path`. By default, pyarmor will
    # search dynamic library `_pytransform` in the same path with entry script
    # `hello.py`. When `hello.py` resides in the `library.zip`, it's failed to
    # load. By setting `runtime-path` to empty string, tell pyarmor search
    # dynamic library in the same path of `library.zip`
    #
    ./pyarmor config --runtime-path=''

    # Obfuscate scripts, after this command,
    #
    #   All the obfuscated scirpts in the path `dist`
    #   All the runtime files in the path `runtimes`
    #
    ./pyarmor build

    # After that, copy all the obfuscated python scripts to distpath of pyinstaller
    # copy all the runtime files to distpath of pyinstaller
    cp dist/*.py /path/to/pyinstaller/distpath
    cp runtimes/* /path/to/pyinstaller/distpath   

    # Finally package all the files in the distpath by pyinstaller
```
The key is that all the runtime files include `_pytransform.so` should be in the same path of entry execute file (hello.exe) generated by pyinstaller.

Here is an examples about py2exe:
https://github.com/dashingsoft/pyarmor/tree/master/src/examples#work-with-py2exe

**Comment by zacharyzugg** [comment]
Running into similar issue.  I've used pyinstaller to build the package.  In the package all pyarmor runtime files are along side my executable:

├myproject_dir
    ├── MyExecutable
    ├── _pytransform.dylib
    ├── license.lic
    ├── pytransform.key
    ├── pytransform.py

When I run my executable I get the following error:
Protection Fault: [Errno 2] No such file or directory: '_pytransform.dylib'

When I REMOVE _pytransform.dylib from the package I get:
Could not find "/myproject/dist/myproject_dir/_pytransform.dylib"

It seems to be aware of _pytransform.dylib  since when I remove it the error changes.

I've tried `def pyarmor_runtime(path="./")`  but I get:
Could not find "./_pytransform.dylib"

Everything works fine in the obfuscated dir prior to pyinstaller build
